web/src/containers/activity/Expenses.js

```diff
@@ -24,26 +26,172 @@ const EXPENSE_CATEGORIES = [
+  <div className="mb1 h5 flex justify-between">
+    <button
+      type="button"
+      onClick={toggleForm}
+      className="btn btn-no-focus p1 col-12 left-align bg-blue-light rounded-left"
+    >
+      <div className="flex items-center justify-between">
+        <div className="col-5 truncate">
+          {idx + 1}. <strong>{expense.category}</strong>
+        </div>
+        {years.map(year => (
+          <div key={year} className="col-2 truncate">
+            {year}:{' '}
+            <span className="bold mono">
+              {formatMoney(expense.years[year])}
+            </span>
+          </div>
+        ))}
+      </div>
+    </button>
+    <button
+      type="button"
+      onClick={handleDelete}
+      className="btn btn-no-focus p1 bg-blue-light rounded-right"
+    >
+      ✗
+    </button>
+  </div>
+);
+
+ExpenseEntry.propTypes = {
+  expense: PropTypes.object.isRequired,
+  idx: PropTypes.number.isRequired,
+  years: PropTypes.array.isRequired,
+  handleDelete: PropTypes.func.isRequired,
+  toggleForm: PropTypes.func.isRequired
+};
+
+const ExpenseForm = ({
+  expense,
+  idx,
+  years,
+  handleChange,
+  handleYearChange,
+  handleDelete
+}) => (
+  <div className="mt2 mb3">
+    <div className="mb3 md-flex">
+      <Label>Category</Label>
+      <Select
+        name={`expense-${expense.key}-category`}
+        options={EXPENSE_CATEGORIES}
+        label="Expense category"
+        hideLabel
+        value={expense.category}
+        onChange={handleChange(idx, 'category')}
+        wrapperClass="md-col-6"
+      />
+    </div>
+    <div className="mb3 md-flex">
+      <Label>Description</Label>
+      <Textarea
+        name={`expense-${expense.key}-desc`}
+        label="Describe the expense"
+        hideLabel
+        rows="3"
+        value={expense.desc}
+        onChange={handleChange(idx, 'desc')}
+        wrapperClass="md-col-6"
+      />
+    </div>
+    {years.map(year => (
+      <div key={year} className="mb3 md-flex">
+        <Label>{year} Cost</Label>
+        <DollarInput
+          name={`expense-${expense.key}-${year}-cost`}
+          label={`Cost for ${year}`}
+          hideLabel
+          value={expense.years[year]}
+          onChange={handleYearChange(idx, year)}
+          wrapperClass="md-col-3"
+        />
+      </div>
+    ))}
+    <div>
+      <Btn kind="outline" extraCss="px1 py-tiny h5" onClick={handleDelete}>
+        Remove expense
+      </Btn>
+    </div>
+  </div>
+);
+
+ExpenseForm.propTypes = {
+  expense: PropTypes.object.isRequired,
+  idx: PropTypes.number.isRequired,
+  years: PropTypes.array.isRequired,
+  handleChange: PropTypes.func.isRequired,
+  handleYearChange: PropTypes.func.isRequired,
+  handleDelete: PropTypes.func.isRequired
+};
+
+const getLastExpense = expenses =>
+  expenses.length ? expenses[expenses.length - 1].key : null;
+
 class Expenses extends Component {
+  static getDerivedStateFromProps(props, state) {
```

Comment:
It looks like this is never used, but it'd almost satisfy [Laura's comment](https://github.com/18F/cms-hitech-apd/pull/955#pullrequestreview-148177788).  It looks like this code is intended to default the **last** expense to opened.

Reply:
this is used -- when a new expense gets added, it adds that new expense key to the state and defaults it to be open